TaskWithInheritanceAndPolymorphism/TaskWithInheritanceAndPolymorphism.cpp

```diff
@@ -0,0 +1,139 @@
+﻿#include <iostream>
+#include <memory>
+#include <vector>
+
+class Figure
```

Comment:
Please declare a Figure as a template class like:
`template <typename T>
class Figure`

Read how to forbid instantiating of this class using non-numeric types.

Reply:
Done with another way